viam-cartographer/cartographer_patches/carto.patch

```diff
@@ -0,0 +1,304 @@
++  absl::utility
+ )
++if (TARGET absl::log_internal_check_op)
++  target_link_libraries(${PROJECT_NAME} PUBLIC absl::log_internal_check_op)
++endif()
+ if (NOT WIN32)
+   target_link_libraries(${PROJECT_NAME} PUBLIC pthread)
+ endif()
+-if(${BUILD_GRPC})
+-  target_link_libraries(${PROJECT_NAME} PUBLIC grpc++)
+-  target_link_libraries(${PROJECT_NAME} PUBLIC async_grpc)
+-endif()
+-if(${BUILD_PROMETHEUS})
+-  target_link_libraries(${PROJECT_NAME} PUBLIC ${ZLIB_LIBRARIES})
+-  target_link_libraries(${PROJECT_NAME} PUBLIC prometheus-cpp-core)
+-  target_link_libraries(${PROJECT_NAME} PUBLIC prometheus-cpp-pull)
+-  target_compile_definitions(${PROJECT_NAME} PUBLIC USE_PROMETHEUS=1)
+-endif()
+ 
+ set(TARGET_COMPILE_FLAGS "${TARGET_COMPILE_FLAGS} ${GOOG_CXX_FLAGS}")
+ set_target_properties(${PROJECT_NAME} PROPERTIES
+   COMPILE_FLAGS ${TARGET_COMPILE_FLAGS})
+ 
+-set(TEST_LIB
+-  cartographer_test_library
+-)
+-add_library(${TEST_LIB} ${TEST_LIBRARY_HDRS} ${TEST_LIBRARY_SRCS})
+-target_include_directories(${TEST_LIB} SYSTEM PRIVATE
+-  "${GMOCK_INCLUDE_DIRS}")
+-# Needed for dynamically linked GTest on Windows.
+-if (WIN32)
+-  target_compile_definitions(${TEST_LIB} PUBLIC -DGTEST_LINKED_AS_SHARED_LIBRARY)
+-endif()
+-target_link_libraries(${TEST_LIB} PUBLIC ${GMOCK_LIBRARY})
+-target_link_libraries(${TEST_LIB} PUBLIC ${PROJECT_NAME})
+-set_target_properties(${TEST_LIB} PROPERTIES
+-  COMPILE_FLAGS ${TARGET_COMPILE_FLAGS})
+-
+-foreach(ABS_FIL ${ALL_TESTS})
+-  file(RELATIVE_PATH REL_FIL ${PROJECT_SOURCE_DIR} ${ABS_FIL})
+-  get_filename_component(DIR ${REL_FIL} DIRECTORY)
+-  get_filename_component(FIL_WE ${REL_FIL} NAME_WE)
+-  # Replace slashes as required for CMP0037.
+-  string(REPLACE "/" "." TEST_TARGET_NAME "${DIR}/${FIL_WE}")
+-  google_test("${TEST_TARGET_NAME}" ${ABS_FIL})
+-  if(${BUILD_GRPC})
+-    target_link_libraries("${TEST_TARGET_NAME}" PUBLIC grpc++)
+-    target_link_libraries("${TEST_TARGET_NAME}" PUBLIC async_grpc)
+-  endif()
+-  if(${BUILD_PROMETHEUS})
+-    target_link_libraries("${TEST_TARGET_NAME}" PUBLIC ${ZLIB_LIBRARIES})
+-    target_link_libraries("${TEST_TARGET_NAME}" PUBLIC prometheus-cpp-core)
+-    target_link_libraries("${TEST_TARGET_NAME}" PUBLIC prometheus-cpp-pull)
+-  endif()
+-  target_link_libraries("${TEST_TARGET_NAME}" PUBLIC ${TEST_LIB})
+-endforeach()
+-
+ # Add the binary directory first, so that port.h is included after it has
+ # been generated.
+ target_include_directories(${PROJECT_NAME} PUBLIC
+diff --git a/cmake/functions.cmake b/cmake/functions.cmake
+index 8cfc05f..6d849ca 100644
+--- a/cmake/functions.cmake
++++ b/cmake/functions.cmake
+@@ -87,7 +87,7 @@ macro(google_initialize_cartographer_project)
+     google_add_flag(GOOG_CXX_FLAGS "-Werror=reorder")
+     google_add_flag(GOOG_CXX_FLAGS "-Werror=return-type")
+     google_add_flag(GOOG_CXX_FLAGS "-Werror=switch")
+-    google_add_flag(GOOG_CXX_FLAGS "-Werror=uninitialized")
++    # google_add_flag(GOOG_CXX_FLAGS "-Werror=uninitialized")
```

Comment:
[q] why is this being removed?

Reply:
-Werror turns a warning into a (fatal) error. Newer GCC finds uninitialized values here that older GCC didn't, so this causes the entire build to fail on newer linux distros. With this disabled, they become just warnings again.

Reply:
Should this line just be fully removed then?